src/ess/imaging/normalize.py

```diff
@@ -0,0 +1,235 @@
+NormalizedSampleImages = NewType("NormalizedSampleImages", sc.DataArray)
+"""Normalized sample image stack.
+
+~math:`normalized_sample = sample / background * DFactor`.
+"""
+GroupedSampleImages = NewType("GroupedSampleImages", sc.DataArray)
+"""Grouped sample image stack by rotation angle."""
+
+
+def _mean_all_dims(data: sc.Variable) -> sc.Variable:
+    """Calculate the mean of all dimensions one by one to avoid overflow."""
+    if data.shape == ():  # scalar
+        return data
+    return _mean_all_dims(data.mean(dim=data.dims[0]))
+
+
+def average_open_beam_images(open_beam: OpenBeamImageStacks) -> OpenBeamImage:
+    """Average the open beam image stack.
+
+    .. math::
+
+        OpenBeam = mean(open_beam, 'time')
+
+    """
+    return OpenBeamImage(sc.mean(open_beam, dim=TIME_COORD_NAME))
+
+
+def average_dark_current_images(
+    dark_current: DarkCurrentImageStacks,
+) -> DarkCurrentImage:
+    """Average the dark current image stack.
+
+    .. math::
+
+        DarkCurrent = mean(dark_current, 'time')
+
+    """
+    return DarkCurrentImage(sc.mean(dark_current, dim=TIME_COORD_NAME))
+
+
+BackgroundPixelThreshold = NewType("BackgroundPixelThreshold", sc.Variable)
+"""Threshold for the background pixel values."""
+
+DEFAULT_BACKGROUND_THRESHOLD = BackgroundPixelThreshold(sc.scalar(1.0, unit="counts"))
+
+
+def calculate_white_beam_background(
+    open_beam: OpenBeamImage,
+    dark_current: DarkCurrentImage,
+    background_threshold: BackgroundPixelThreshold = DEFAULT_BACKGROUND_THRESHOLD,
+) -> BackgroundImage:
+    """Calculate the background image stack.
+
+    We average the open beam and dark current image stack
+    to create the single background image.
+
+    .. math::
+
+        Background = mean(OpenBeam, 'time') - mean(DarkCurrent, 'time')
+
+
+    Params
+    ------
+    oepn_beam:
+        Open beam image stack.
+
+    dark_current:
+        Dark current image.
+
+    background_threshold:
+        Threshold for the background pixel values.
+        Any pixel values less than ``background_threshold``
+        are replaced with ``background_threshold``.
+        Default is 1.0[counts].
+
+
+    """
+    diff = open_beam - dark_current
+    diff.data = sc.where(
+        diff.data < background_threshold, background_threshold, diff.data
+    )
+    return BackgroundImage(diff)
+
+
+SamplePixelThreshold = NewType("SamplePixelThreshold", sc.Variable)
+"""Threshold for the sample pixel values."""
+
+DEFAULT_SAMPLE_THRESHOLD = SamplePixelThreshold(sc.scalar(0.0, unit="counts"))
+
+
+def cleanse_sample_images(
+    sample_images: SampleImageStacks,
+    dark_current: DarkCurrentImage,
+    sample_threshold: SamplePixelThreshold = DEFAULT_SAMPLE_THRESHOLD,
+) -> CleansedSampleImages:
+    """Cleanse the sample image stack.
+
+    We subtract the averaged dark current image from the sample image stack.
+
+    .. math::
+
+        CleansedSample_{i} = Sample_{i} - mean(DarkCurrent, dim='time')
```

Comment:
Same here: also add (in words or in the math) the fact that you are applying a threshold after the operation?